src/server/server.ts

```diff
@@ -753,10 +753,21 @@ namespace ts.server {
     const sys = <ServerHost>ts.sys;
     // use watchGuard process on Windows when node version is 4 or later
     const useWatchGuard = process.platform === "win32" && getNodeMajorVersion() >= 4;
+    const originalWatchDirectory = sys.watchDirectory;

```

Comment:
For type safety, consider `const originalWatchDirectory: ServerHost["watchDirectory"] = sys.watchDirectory.bind(sys);` and call it like a normal function.
But what if `sys.watchDirectory` is absent -- apparently this is optional?

Reply:
+1 for @andy-ms's suggession. 

Also consider adding a comment why we swallow these exceptions e.g. in the Linux case

Reply:
It is optional but tsserver had it like that all the time.